### PR TITLE
updates code/okta_react/ examples for the okta-react 2.x

### PR DESCRIPTION
**Pull request by swiftone**
## Description:
- **What's changed?**
  - Updates the /code/react/okta_react examples to cover okta-react 2.x
  - Also gives both function-based component and class-based component examples
- **Is this PR related to a Monolith release?** 
  - No
### Resolves:

* [OKTA-272787](https://oktainc.atlassian.net/browse/OKTA-272787)

### CODE REVIEW

File: packages/@okta/vuepress-site/code/react/okta_react/index.md [hunk 1]
```diff
@@ -39,31 +39,89 @@ A simple way to add authentication to a React app is using the [Okta Auth JS](/c
 npm install @okta/okta-auth-js --save
 ```
 
-We'll also need `@okta/okta-react` and `react-router-dom` to manage our routes:
+We'll also need `@okta/okta-react` and `react-router-dom` to manage our routes (`@okta/okta-react` can be used to support other router libraries, but `react-router-dom` has pre-existing support).
 ```bash
 npm install @okta/okta-react react-router-dom --save
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/code/react/okta_react/index.md]: Should we split this script to make `react-router-dom` optional, also mention potential npm warning?

Comment (swiftone) [reply packages/@okta/vuepress-site/code/react/okta_react/index.md]: This is a get-started script, so we're showing them the fastest way to get started.  

The optional-ness and warning-if-omitted are covered by okta-react docs, this is more showing them that the library is worthwhile.  (it's also due to be replaced by our guides, but that's a slower process).

File: packages/@okta/vuepress-site/code/react/okta_react/index.md [hunk 2]
```diff
@@ -39,31 +39,89 @@ A simple way to add authentication to a React app is using the [Okta Auth JS](/c
...
+  };
+
+  if (sessionToken) {
+    authService.redirect({ sessionToken });
+    return null;
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <label>
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/code/react/okta_react/index.md]: Two <label> tags instead of one?

Comment (swiftone) [reply packages/@okta/vuepress-site/code/react/okta_react/index.md]: Ah good catch - this was in the original, fixed now.

**Pull request merged**